### PR TITLE
fix: workflow permissions and use inst/WORDLIST

### PR DESCRIPTION
**Pull request by akselthomsen**
Updated the following to make Megalinters `checkov` and `cspell` checks work as intended:

- For `checkov` the individual repos should set the relevant permissions before calling the workflows. This is now reflected in the snippet in the readme.
- Unnecessary permission `issue: write` has been removed from he MegaLinter workflow.
- Added a second `cspell` configuration that adds the `inst/WORDLIST` file used by `pre.commit` to the dictionary of approved words.
- Changed Megalinter workflow to use the above configuration if `inst/WORDLIST` exists, and otherwise use the old one.

Note: It is hard to test the workflows before approving, since they point to a main branch, wherein the files do not exist yet. I have made a previous test run on {whirl} that shows it is working:

https://github.com/NovoNordisk-OpenSource/whirl/actions/runs/12432723171/job/34712764428?pr=126

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ It contains the following workflows:
 2. [R-CMD-check NN versions](.github/workflows/check_nn_versions.yaml): Same as 1., but uses the R version and packages available as per given lock dates.
 3. [Test coverage](.github/workflows/coverage.yaml): Derives test coverage for the package and publishes a summary table to the pull request.
 4. [pkgdown](.github/workflows/pkgdown.yaml): Renders and publishes a `pkgdown` website for your package (to your `gh-pages` branch). For a pull request the page is published to `{base url of package}/dev/{pr number}`, and a link to this development webpage is posted as a comment to your pull request.
-5. [megalinter](.github/workflows/megalinter.yaml): Lints your entire project using the [megalinter](https://megalinter.io/) tool.
+5. [megalinter](.github/workflows/megalinter.yaml): Lints your entire project using the [megalinter](https://megalinter.io/) tool. Note that for the [cspell](https://github.com/streetsidesoftware/cspell) linter words in `inst/WORDLIST` are automatically added as a dictionry if the file exists.
```
Comment (vladimirobucina) [inline README.md:11]: just a small typo here, dictionry 😄

**Review by vladimirobucina — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (akselthomsen) [reply README.md:11]: Good catch! And before I have even opened the PR :)

Comment (vladimirobucina) [reply README.md]: this is how we roll 🥳

**Review by vladimirobucina — APPROVED** [review]
Everything looks good now.

**Pull request merged**